src/super_gradients/training/models/segmentation_models/unet/unet.py

```diff
@@ -0,0 +1,231 @@
+        :param align_corners: align_corners arg of segmentation and auxiliary heads.
+        :param backbone_params: params to build a `UNetBackboneBase`, include the following keys:
+            - strides_list: List[int], list of stride per stage.
+            - width_list: List[int], list of num channels per stage.
+            - num_blocks_list: List[int], list of num blocks per stage.
+            - block_types_list: List[Union[DownBlockType, int]], list of block types per stage.
+            - is_out_feature_list: List[bool], list of flags whether stage features should be an output.
+            - in_channels: int, num channels of the input to the backbone module.
+            - block_params: dict, argument to be passed to the block types constructors. i.e for `RegnetXStage`
+                block_params should include bottleneck_ratio, group_width and se_ratio.
+        :param context_module_params: params to build the context module currently only `ASPP` is supported, include the
+            following keys:
+            - dilation_list: List[int], dilation rates for the ASPP module.
+        :param decoder_params: params to build a `Decoder`, include the following keys:
+            - up_block_repeat_list: List[int], num of blocks per decoder stage, the `block` implementation depends on
+                the up-block type.
+            - skip_expansion: float, skip expansion ratio value, before fusing the skip features from the encoder with
+                the decoder features, a projection convolution is applied upon the encoder features to project the
+                num_channels by skip_expansion.
+            - decoder_scale: float, num_channels width ratio between encoder stages and decoder stages.
+            - up_blocks: List[Type[AbstractUpFuseBlock]], list of AbstractUpFuseBlock types.
+            - is_skip_list: List[bool], List of flags whether to use feature-map from encoder stage as skip connection
+                or not.
+        :param aux_heads_params: params to initiate auxiliary heads, include the following keys:
+            - use_aux_list: List[bool], whether to append to auxiliary head per encoder stage.
+            - aux_heads_factor: List[int], Upsample factor per encoder stage.
+            - aux_hidden_channels: List[int], Hidden num channels before last classification layer, per encoder stage.
+            - aux_out_channels: List[int], Output channels, can be refers as num_classes, of auxiliary head per encoder
+                stage.
+        :param dropout: dropout probability of segmentation and auxiliary heads.
+        """
+        super().__init__(use_aux_heads=use_aux_heads)
+        self.num_classes = num_classes
+        # Init Backbone
+        backbone = UNetBackboneBase(**backbone_params)
+        # Init Context Module
+        context_module = nn.Identity() if context_module is None else build_context_module(
```

Comment:
this piece of code is redundant. use `@resolve_parameter(context_module, ContextModuleFactory)` 
in_channels is known from backbone_params and can be defined in YAML

Reply:
Done [6bf7a2b](https://github.com/Deci-AI/super-gradients/pull/401/commits/6bf7a2bb76c41d8af67f4390f2da9bc9c0d59ddb)